File: mobile/lib/features/discovery/domain/college_model.dart
class College {
  final String id;
  final String name;
  final String slug;
  final String city;
  final String state;
  final String? logo;
  final bool isActive;

  College({
    required this.id,
    required this.name,
    required this.slug,
    required this.city,
    required this.state,
    this.logo,
    required this.isActive,
  });

  factory College.fromJson(Map<String, dynamic> json) {
    return College(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      slug: json['slug'] ?? '',
      city: json['city'] ?? '',
      state: json['state'] ?? '',
      logo: json['logo'],
      isActive: json['isActive'] ?? true,
    );
  }
}

File: backend/src/modules/colleges/college.model.js
const mongoose = require('mongoose');
const slugify = require('slugify');

const collegeSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
  },
  slug: {
    type: String,
    unique: true,
    index: true,
  },
  city: {
    type: String,
    required: true,
  },
  state: {
    type: String,
    required: true,
  },
  location: {
    type: {
      type: String,
      enum: ['Point'],
      default: 'Point',
    },
    coordinates: {
      type: [Number], // [longitude, latitude]
      required: true,
    }
  },
  logo: String,
  isActive: {
    type: Boolean,
    default: true,
  }
}, {
  timestamps: true,
});

// Explicit geospatial index for nearby college search
collegeSchema.index({ location: '2dsphere' });

// Middleware to auto-generate slug before saving
collegeSchema.pre('save', function () {
  if (this.isModified('name')) {
    this.slug = slugify(this.name, { lower: true, strict: true });
  }
});

module.exports = mongoose.model('College', collegeSchema);
